### PR TITLE
#463 Frontend can fetch a subset of CaseReportsForListing.

### PR DESCRIPTION
**Pull request by woksin**
Added a method in CaseReportsForListing that fetches a maximum amount, limit, of reports from its collection either from the front or back specified by a boolean argument.

Re-added the obsolete method from CaseReportController that retrieves the reports from CaseReports collection. It's nice to have for testing.

The frontend stuff with TS and Angular frustrates me, my efficiency is drastically reduced when working with the frontend stuff. We need someone to look at the frontend issues for VolunteerReporting.

### CODE REVIEW

File: Source/VolunteerReporting/Read/Read.csproj [hunk 1]
```diff
@@ -13,5 +13,10 @@
     <PackageReference Include="doLittle.Events.Processing" Version="2.0.0-alpha2.194" />
     <PackageReference Include="doLittle.Runtime.Events" Version="2.0.0-alpha2.70" />
   </ItemGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.AspNetCore.Mvc.Core">
+      <HintPath>..\..\..\..\..\..\..\..\Program Files\dotnet\sdk\NuGetFallbackFolder\microsoft.aspnetcore.mvc.core\2.0.2\lib\netstandard2.0\Microsoft.AspNetCore.Mvc.Core.dll</HintPath>
```
Comment (woksin) [inline Source/VolunteerReporting/Read/Read.csproj]: Weird, the Read project should not have use of this.

File: Source/VolunteerReporting/Web.Angular/src/app/case-report/case-report-list/case-report-list.component.ts [hunk 2]
```diff
@@ -33,10 +34,19 @@ export class CaseReportListComponent implements OnInit {
...
+        "timestamp", "location"
+    ];
+
     constructor(
         private caseReportService: AggregatedCaseReportService,
-        private service: ReportService
-    ) { this.maxReports = 1000 }
+        private service: ReportService,
+        private caseReportExporter: CaseReportExporter
+    ) { this.maxReports = 10 }
```
Comment (woksin) [inline Source/VolunteerReporting/Web.Angular/src/app/case-report/case-report-list/case-report-list.component.ts]: Should change this to a bigger number, 100 for example.

File: Source/UserManagement/Web/Controllers/UserController.cs [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 namespace Web
 {
-    [Route("api/usermanagement/")]
+    [Route("api/staffuser")]
```
Comment (Velcrow81) [inline Source/UserManagement/Web/Controllers/UserController.cs]: Restfull api naming is typically plural, so staffusers is more the norm :)

File: Source/VolunteerReporting/Read/CaseReportsForListing/CaseReportsForListing.cs [hunk 4]
```diff
@@ -26,6 +33,26 @@ public CaseReportsForListing(IMongoDatabase database)
             return await list.ToListAsync();
         }
 
+        public async Task<IEnumerable<CaseReportForListing>> GetLimitAsync(int limit, Boolean last)
+        {
+            
+            var filter = Builders<CaseReportForListing>.Filter.Empty;
+
+            // TODO: Can we assume that this explicit conversion is safe?
```
Comment (Velcrow81) [inline Source/VolunteerReporting/Read/CaseReportsForListing/CaseReportsForListing.cs]: Don't know

File: Source/VolunteerReporting/Read/CaseReportsForListing/CaseReportsForListing.cs [hunk 5]
```diff
@@ -26,6 +33,26 @@ public CaseReportsForListing(IMongoDatabase database)
...
+            var totalDocuments = (int)await _collection.CountAsync(filter);
+
+            limit = (limit > totalDocuments) ? totalDocuments : limit;
+
+            var options = new FindOptions<CaseReportForListing>()
+            {
+                Skip = last ? totalDocuments - limit : 0,
```
Comment (Velcrow81) [inline Source/VolunteerReporting/Read/CaseReportsForListing/CaseReportsForListing.cs]: Is this correct? If you don't want the first 100, but the following 100 in a collection of 1000, you enter 100 and last = true. That means 1000 - 100 = 900, i.e. you skip the first 900, not just 100. If you enter 900, then you skip the first 100, but it sounds strange with a limit of 900 to get 100. Maybe limit is not right word? Am I missing something?

File: Source/VolunteerReporting/Web.Angular/src/app/case-report/case-report-list/case-report-list.component.ts [hunk 6]
```diff
@@ -33,10 +34,19 @@ export class CaseReportListComponent implements OnInit {
 
     basicFilter: string = "all";
     
+    fields: Array<string> = [
+        "id", "status", "dataCollectorId", "dataCollectorDisplayName",
+        "healthRiskId", "healthRisk", "message",
+        "numberOfFemalesOver5", "numberOfFemalesUnder5",
```
Comment (Velcrow81) [inline Source/VolunteerReporting/Web.Angular/src/app/case-report/case-report-list/case-report-list.component.ts]: I have just renamed the age properties. Ensure that these are correct after merge with my PR :)

File: Source/UserManagement/Web/Controllers/UserController.cs (hunk 3 above)
Comment (woksin) [reply Source/UserManagement/Web/Controllers/UserController.cs]: Okay, I will keep that in mind then.

File: Source/VolunteerReporting/Read/CaseReportsForListing/CaseReportsForListing.cs (hunk 5 above)
Comment (woksin) [reply Source/VolunteerReporting/Read/CaseReportsForListing/CaseReportsForListing.cs]: Well, limit is just the max amount of reports you want to fetch from the collection. And depending on what you want you can retrieve the limit last elements or the limit first

Comment (woksin) [reply Source/VolunteerReporting/Read/CaseReportsForListing/CaseReportsForListing.cs]: In my implementation here there is not really room for getting the "in between" documents...
It all very much depends on what you want the system to do really.

Comment (woksin) [reply Source/VolunteerReporting/Read/CaseReportsForListing/CaseReportsForListing.cs]: Limit= 900 and last = true would fetch the last 900 documents, skipping the first Collection.Size - 900 documents.

File: Source/VolunteerReporting/Web.Angular/src/app/case-report/case-report-list/case-report-list.component.ts (hunk 6 above)
Comment (woksin) [reply Source/VolunteerReporting/Web.Angular/src/app/case-report/case-report-list/case-report-list.component.ts]: Will do

**Pull request merged**